nexus/tests/integration_tests/certificates.rs

```diff
@@ -599,11 +600,7 @@ async fn test_silo_certificates() {
         );
     if let oxide_client::Error::CommunicationError(error) = error {
         assert!(error.is_connect());
-        assert!(
-            error.to_string().contains("invalid peer certificate")
-                || error.to_string().contains("self-signed certificate")
-                || error.to_string().contains("self signed certificate")
-        );
+        assert!(error.chain().to_string().contains("self-signed certificate"));
```

Comment:
@jmpesp I think this was you; does this new check suffice?

Reply:
It's been so long I don't remember doing this haha!

I'm not sure this check is right: it looks like this part of the test is meant to check that a client can't connect using the wrong certificate, and this check is asserting it won't trust a self-signed certificate. That's not quite the same thing. I think we need either [`add_root_certificate`](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.add_root_certificate) or [`danger_accept_invalid_certs`](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.danger_accept_invalid_certs) in order to test that the client accepts the self-signed cert but fails due to the incorrect certificate being used.

Reply:
https://github.com/oxidecomputer/omicron/issues/6716